core/eval.go

```diff
@@ -1013,6 +1049,15 @@ func EvalAndRespond(cmds RedisCmds, c *Client) {
 	buf := bytes.NewBuffer(response)
 
 	for _, cmd := range cmds {
+		// Check if the command has been authenticated
+		if cmd.Cmd != AuthCmd && !c.Session.IsActive() {
+			// TODO: Check required response
+			log.Println("Session not active for CMD:", cmd.Cmd)
+			if _, err := c.Write(RESP_AUTH_FAILURE); err != nil {
+				log.Panic(err)
```

Comment:
panic is not a good idea. Panic will terminate the server, which is bad.

Reply:
Yes, oversight on my part. Removed the panic and logging the failure now.